templates/recipe_detail.html

```diff
@@ -1,8 +1,19 @@
 By <a href="/author/{{recipe.author.id}}/">{{recipe.author.name}}</a>
 
-<body style="background-color: bisque;"></body>
+<body style="background-color: bisque"></body>
 
-{{recipe.title}} 
+{{recipe.title}}
+<br />
 {{recipe.description}}
+<br />
 {{recipe.time_required}}
-{{recipe.instructions}}
\ No newline at end of file
+<br />
+{{recipe.instructions}}
+<br />
+{% if request.user == recipe.author.user %}
+
+You must be the author or an administrator to edit this post
+<a href="/recipe/{{ recipe.id }}/edit/">Edit This Recipe</a>
+{% endif %}
+Put this post in your Recipe Box
+<a href="/recipe/{{ recipe.id }}/favorite/">Add to My Recipe Box</a>
```

Comment:
This button/link should not be visible when the user is logged out. 

What variable could we check to see if the user is logged in or not? How could we display this button/link based upon that information?